### PR TITLE
[Pallas] Allow setting FlashAttention's causal mask

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request channels the causal mask to our wrapper.

Test Plan:
PJRT_DEVICE=TPU python test/test_pallas.py -v -k test_flash_attention_wrapper_causal

### CODE REVIEW

File: test/test_pallas.py [hunk 1]
```diff
@@ -224,6 +224,30 @@ def attention(q, k, v):
...
+      attn_weight = nn.functional.softmax(attn_weight, dim=-1)
+      attn_output = attn_weight @ v
+      return attn_output
+
+    q = torch.randn(3, 2, 128, 4).to("xla")
+    k = torch.randn(3, 2, 128, 4).to("xla")
+    v = torch.randn(3, 2, 128, 4).to("xla")
+
+    # The causal mask is turned on by default in the wrapper.
+    # It masks out the top right triangle of the attention matrix, therefore it speeds up the compute but also changes the output.
```
Comment (JackCaoG) [inline test/test_pallas.py:245]: what do you mean by `also changes the output`?

**Review by JackCaoG — APPROVED** [review]

File: test/test_pallas.py (hunk 1 above)
Comment (alanwaketan) [reply test/test_pallas.py:245]: The output will be different from the eager attention output, which is reflected in the test.

**Comment by alanwaketan** [comment]
Thanks Jack for approving.
